config/m4/module.m4

```diff
@@ -0,0 +1,437 @@
+
+             m4_foreach([func], [$3],
+                 [ dnl have to clean func check cache to allov multiple checks
+                  AS_VAR_PUSHDEF([func_cache], [ac_cv_search_[]func])dnl
+                  unset func_cache
+                  mod_func_name=func
+                  AC_SEARCH_LIBS([$mod_func_name], [$1], [], [mod_libs_happy=no], [$6])
+                  AS_VAR_POPDEF([func_cache])])dnl
+
+             LDFLAGS=$mod_libs_saved_LDFLAGS
+
+             AS_IF([test $mod_libs_happy = yes], [$4], [$5])
+         ])
+
+#
+# Check module API
+#
+# Usage: _CHECK_MODULE_API([incdir], [libdir], [if-found], [if-not-found],
+#                          [[header], [decls], [includes], [libs], [funcs], [otherlibs]])
+#
+AC_DEFUN([_CHECK_MODULE_API],
+         [
+             _CHECK_MODULE_DEFVAR([$5])
+
+             mod_api_happy=no
+
+             _CHECK_MODULE_HEADER(mod_header, [$1],
+                     [_CHECK_MODULE_DECLS(mod_decls, [$1],
+                              [_CHECK_MODULE_LIBS(mod_libs, [$2], mod_funcs,
+                                      [
+                                          $3
+                                          mod_api_happy=yes
+                                      ],
+                                      [], [mod_ldflags])],
+                                      [], mod_includes)])
+
+             _CHECK_MODULE_UNDEFVAR
+
+             AS_IF([test $mod_api_happy = no], [$4])
+         ])
+
+#
+# Check module package
+#
+# Usage: _CHECK_MODULE_PACKAGE([pkgid], [prefix], [libdir], [static], [if-found], [if-not-found],
+#                              [[header], [decls], [includes], [libs], [funcs], [ldflags]])
+#
+# DO NOT!!!!! use macro substitution for 'prefix' var of  _CHECK_MODULE_PACKAGE
+# because m4 can't expand macro in PKG_CHECK_MODULES
+#
+AC_DEFUN([_CHECK_MODULE_PACKAGE],
+         [
+             mod_pkg_saved_PKG_CONFIG_PATH="$PKG_CONFIG_PATH"
+             mod_package_happy=yes
+
+             AS_IF([test "x$1" = "x"], [mod_package_happy=no])
+
+             AS_IF([test "x$3" != "x"],
+                   [
+                       _CHECK_MODULE_IS_DIR([$3/pkgconfig],
+                                            [export PKG_CONFIG_PATH="$3/pkgconfig:$PKG_CONFIG_PATH"],
+                                            [mod_package_happy=no])
+                   ])
+
+             AS_IF([test $mod_package_happy = yes],
+                   [PKG_CHECK_MODULES$4([$2], [$1], [], [mod_package_happy=no])])
+
+             export PKG_CONFIG_PATH="$mod_pkg_saved_PKG_CONFIG_PATH"
+
+             AS_IF([test $mod_package_happy = yes],
+                   [
+                       _CHECK_MODULE_API([$[$2_CFLAGS]], [$[$2_LIBS]],
+                                         [], [mod_package_happy=no], [$7])
+                   ])
+             
+             AS_IF([test $mod_package_happy = yes], [$5], [$6])
+         ])
+
+# Check if module available
+# Result: set variables prefix_CPPFLAGS and prefix_LIBS
```

Comment:
We use prefix_LIBS and prefix_LDFLAGS when configuring some modules.
E.g. prefix_LIBS contains list of libraries: -lmylib, prefix_LDFLAGS contains path to library: -L$path_to_my_lib.
Perhaps this is a better approach. What do you think about it?

Reply:
is there any case when LIBS is used without LDFLAGS?

Reply:
I used same as for PKG_CHECK_MODULES variables to store results

Reply:
> is there any case when LIBS is used without LDFLAGS?

Actually, prefix_LDFLAGS usually is added to module_la_LDFLAGS/bin_LDFLAGS, prefix_LIBS usually is added to module_la_LIBADD/bin_LIBADD in .am files.

Reply:
the problem is in using PKG_CHECK_MODULES macro to identify module: it uses LIBS/CFLAGS to store results. I used same way to unify results variables

Reply:
I see...

(... the macro will set up two variables ... _CFLAGS and _LIBS. The names of these variables can be somewhat misleading, since the former will generally provide the flags to pass to the preprocessor ... such as include paths and macro definitions, and the latter will provide the library paths as well as the libraries themselves ...)

Therefore, it won't be trivial to rewrite existing configs.